### PR TITLE
ensure unstructured resources in kubectl get

### PR DESCRIPTION
**Pull request by juanvallejo**
Related downstream issue: https://github.com/openshift/origin/issues/16064

Ensure we are dealing with unstructured objects before attempting to
compose into an unstructured list.

`$ kubectl get ...` [assumes all resources returned from a resource builder are unstructured objects](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/get.go#L355). This leads to a panic when dealing with non-unstructured objects.
This patch ensures we are dealing with unstructured objects before attempting to
compose into an unstructured list.

**Release note**:
```release-note
NONE
```

cc @fabianofranz @kubernetes/sig-cli-misc

### CODE REVIEW

**Comment by mengqiy** [comment]
@juanvallejo Can you add a test to prevent regression?

File: pkg/kubectl/cmd/get_test.go [hunk 1]
```diff
@@ -575,6 +575,83 @@ func TestGetListComponentStatus(t *testing.T) {
...
+
+	if len(buf.String()) == 0 {
+		t.Error("unexpected empty output")
+	}
+
+	actual := tf.Printer.(*testPrinter).Objects
+	fn := func(obj runtime.Object) unstructured.Unstructured {
+		data, err := runtime.Encode(api.Codecs.LegacyCodec(schema.GroupVersion{Version: "v1"}), obj)
+		if err != nil {
+			panic(err)
```
Comment (mengqiy) [inline pkg/kubectl/cmd/get_test.go]: Don't panic in the test, use `t.Fatal` or `t.Error`.

File: pkg/kubectl/cmd/get_test.go [hunk 2]
```diff
@@ -575,6 +575,83 @@ func TestGetListComponentStatus(t *testing.T) {
...
+		}
+		out := &unstructured.Unstructured{Object: make(map[string]interface{})}
+		if err := encjson.Unmarshal(data, &out.Object); err != nil {
+			panic(err)
```
Comment (mengqiy) [inline pkg/kubectl/cmd/get_test.go]: Don't panic in the test, use `t.Fatal` or `t.Error`.

**Review by mengqiy — COMMENTED** [review]
Thanks for adding test

**Comment by juanvallejo** [comment]
@mengqiy comments addressed, thanks

**Pull request merged**